akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowThrottleSpec.scala

```diff
@@ -226,10 +237,24 @@ class FlowThrottleSpec extends StreamSpec("""
       downstream.cancel()
     }
 
-    "send elements downstream as soon as time comes" taggedAs GHExcludeTest in assertAllStagesStopped {
-      val probe = Source(1 to 10).throttle(4, 500.millis, 0, _ => 2, Shaping).runWith(TestSink.probe[Int]).request(5)
-      probe.receiveWithin(600.millis) should be(Seq(1, 2))
-      probe.expectNoMessage(100.millis).expectNext(3).expectNoMessage(100.millis).expectNext(4).cancel()
+    "send elements downstream as soon as time comes" in assertAllStagesStopped {
+      val throttleInterval = 300.millis
+      val elementsAndTimestampsMs = Source(1 to 10)
+        .throttle(4, throttleInterval, _ => 2)
+        .runFold(Nil: List[(Long, Int)]) { (acc, n) =>
+          (System.nanoTime() / 1000000, n) :: acc
+        }
+        .futureValue(timeout(5.seconds))
+        .reverse
+
+      val startMs = elementsAndTimestampsMs.head._1
+      val elemsAndTimeFromStart = elementsAndTimestampsMs.map { case (ts, n) => (ts - startMs, n) }
+      val perThrottleInterval = elemsAndTimeFromStart.groupBy {
+        case (fromStart, _) => fromStart / throttleInterval.toMillis
+      }
+      withClue(perThrottleInterval) {
+        perThrottleInterval.forall { case (_, entries) => entries.size == 2 } should ===(true)
```

Comment:
This needs some more thoughts. It failed several times when I tried locally.

```
[info]   HashMap(0 -> List((0,1), (151,2)), 1 -> List((301,3), (449,4), (599,5)), 2 -> List((750,6), (899,7)), 3 -> List((1051,8), (1199,9)), 4 -> List((1350,10))) false did not equal true (FlowThrottleSpec.scala:256)
```

```
[info]   HashMap(0 -> List((0,1), (151,2)), 1 -> List((301,3), (449,4), (599,5)), 2 -> List((751,6)), 3 -> List((901,7), (1050,8)), 4 -> List((1200,9), (1349,10))) false did not equal true (FlowThrottleSpec.scala:256)
```

Reply:
Hmm, I'm surprised that they are spread out rather than come in bursts every throttle interval, I thought I got that right when working on it, but perhaps I broke it again when I switched to the auto-calculated bursts. 

Having a burst size of 2 should make sure they always come close to each other, given that the source is pretty much immediate, right?

Reply:
(Maybe 300ms is too tight as well)

Reply:
Tried the max burst idea without success. A longer time period and only one element per period seems to work more stable on my machine at least.